wocg/main.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+
+import django
+django.setup()
+
+import os
+import re
+import logging
+
+from django.conf import settings
+from weblate.trans.models import Project, SubProject
+
+from .manifest import get_translatable_addons
+
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+ch.setFormatter(logging.Formatter('%(asctime)s - %(levelname)s - %(message)s'))
+logger.addHandler(ch)
+
+GIT_URL_RE = re.compile(r"git@.*:.*\/.*")
+
+
+def _get_main_subproject(project):
+    main_subproject = False
+    for sub_project in project.subproject_set.all():
+        if GIT_URL_RE.match(sub_project.repo):
+            main_subproject = sub_project
+            break
+    return main_subproject
+
+
+def _get_all_subprojects_slug(project):
+    sub_projects_slug = []
+    for sub_project in project.subproject_set.all():
+        sub_projects_slug.append(sub_project.slug)
+    return sub_projects_slug
+
+
+def _get_slug_name(project, addon):
+    return "%s-%s" % (project.slug, addon)
+
+
+def _get_addon_name(component):
+    return component.slug.split('-')[-1]
+
+
+def main():
+    all_projects = Project.objects.prefetch_related('source_language')
+
+    data_dir = settings.DATA_DIR
+    svn_dir = os.path.join(data_dir, 'vcs')
+
+    for project in all_projects:
+        logger.info('WOCG: Begin generation for project %s', project.name)
+        main_subproject = _get_main_subproject(project)
+        if not main_subproject:
+            logger.info('WOCG: Main component not found for project %s'% project.name)
+            continue
+        logger.info('WOCG: Main component found for project %s : %s' % (
+            project.name, main_subproject.name))
+        repo = 'weblate://%s/%s' % (project.slug, main_subproject.slug)
+        main_subproject_addon_name = _get_addon_name(main_subproject)
+        repo_path = os.path.join(svn_dir, project.slug, main_subproject.slug)
+        os.chdir(repo_path)
+        existing_sub_projects_slug = _get_all_subprojects_slug(project)
+        addons = get_translatable_addons()
```

Comment:
may be a bit more robust is starting from the filemask of the main component which should match `r"^(?P<addons_dir>.*)/(?P<addon_name>.*?)/i18n/\*\.po$"` and then listing `addons_dir` to find addons to translate (no need to look at manifest etc).